scheduler/core/components/optimizer/greedymax.py

```diff
@@ -424,9 +439,17 @@ def output_plans(self, plans: Plans) -> None:
             obs_order = timeline.get_observation_order()
             for idx, start_time_slot, end_time_slot in obs_order:
                 if idx > -1:
-                    obs_id = self.obs_group_ids[idx]
+                    # obs_id = self.obs_group_ids[idx]
+                    # TODO: HACK. I don't see an easy way around this since an obs_group_id has no idea it is an
+                    # TODO: UniqueGroupID of an observation group. Maybe we can make obs_in_plan a map from
+                    # TODO: UniqueGroupID to... something, instead of an ObservationID to an ObsPlanData.
+                    unique_group_id = self.obs_group_ids[idx]
+                    obs_id = ObservationID(unique_group_id.id)
 
                     # Add visit to final plan
-                    plans[timeline.site].add(self.obs_in_plan[obs_id]['obs'], self.obs_in_plan[obs_id]['start'],
-                                             start_time_slot, self.obs_in_plan[obs_id]['obs_len'],
-                                             self.obs_in_plan[obs_id]['visit_score'])
+                    obs_in_plan = self.obs_in_plan[obs_id]
```

Comment:
@stroncod Like this is driving me nuts. We can't go from a `UniqueGroupID` to an `ObservationID` although we can go in the other direction since we know that an `Observation` will always be contained in an observation `Group` that has the same ID in its `UniqueGroupID` as is on the `Observation`'s `ObservationID`... however, the converse is not true, and we will have `UniqueGroupID`s that don't correspond to `ObservationID`... I'm wondering if we should just change the `self.obs_in_plan` mapping to use `UniqueGroupID` instead, but then it seems it should have value `Visit` instead of `ObsPlanData` as you say... although the way things are structured, we don't have enough information for that now.